components/user-mgt/org.wso2.carbon.user.mgt/src/main/java/org/wso2/carbon/user/mgt/UserRealmProxy.java

```diff
@@ -117,15 +118,24 @@ public UserRealmProxy(UserRealm userRealm) {
                     flaggedNames[i] = new FlaggedName();
                     flaggedNames[i].setItemName(user);
                     //retrieving the displayName
-                    String displayName = realm.getUserStoreManager().getUserClaimValue(user, DISAPLAY_NAME_CLAIM, null);
-                    int index = user.indexOf(UserCoreConstants.DOMAIN_SEPARATOR);
-                    if (index > 0) {
-                        if (displayName != null) {
+                    // Check whether to use the display name claim when filtering users.
+                    String showDisplayName = IdentityUtil.getProperty(IdentityConstants.SHOW_DISPLAY_NAME);
+                    boolean isShowDisplayNameEnabled = false;
+                    if (StringUtils.isNotBlank(showDisplayName)) {
```

Comment:
No need to check for null. If this value is null or empty the value will be false by default.

Reply:
Fixed.